src/main/java/at/grahsl/kafka/connect/mongodb/MongoDbSinkConnectorConfig.java

```diff
@@ -46,6 +50,13 @@
         WHITELIST
     }
 
+    private static final Pattern CLASS_NAME = Pattern.compile("\\p{javaUnicodeIdentifierStart}\\p{javaUnicodeIdentifierPart}*");
+    private static final Pattern FULLY_QUALIFIED_CLASS_NAME = Pattern.compile("(" + CLASS_NAME + "\\.)*" + CLASS_NAME);
+    private static final Pattern FULLY_QUALIFIED_CLASS_NAME_LIST = Pattern.compile("(" + FULLY_QUALIFIED_CLASS_NAME + ",)*" + FULLY_QUALIFIED_CLASS_NAME);
+    private static final Pattern FIELD = Pattern.compile("(\\*{1,2}|\\p{Alnum}+)");
```

Comment:
Don't want to nit-pick :)
But I think the FIELD pattern is most likely a little too restrictive. Think about JSON fieldnames containing e.g. "-", "_". In general JSON allows almost arbitrary fieldnames and mongodb forbids a few of them such as a "." within the fieldname. Since having a few characters other than p{Alnum} happens in practice a lot we should relax this a bit I guess.

Reply:
You are absolutely right! Way too restrictive! I looked into this earlier today as well. If I read correctly, MongoDB allows any(!) Unicode Character except the dot, which is used as a path separator, and the leading dollar sign, which is used to denote commands.
I did not test, yet. However, it would be very interesting to know whether any intermediary json conversion is as tolerant as mongo. For example, I don't know whether '.' and/or '$' may be escaped, or whether connect itself allows the same character classes as mongo.
I am really learning a lot, here...

Reply:
Unfortunately this one is a really tough one to decide on correctly. JSON itself is happy about a large part of unicode characters to appear in field names/keys. MongoDB (BSON) has the restrictions you are referring to and these can be addressed of course. BUT - the real world punching our faces here - besides JSON, kafka connect is often used based on AVRO messages, a format which comes with a lot more restrictions to fieldnames than JSON or BSON :) For that matter I would suggest to skip/remove any kind of path and thereby field validation for now, until we have a better understanding what makes most sense in this regard.

Reply:
@941design coming back to this PR I just wanted to ask what you think about my suggestion to leave field/path validation out for now. Do you have plans to finish this validation feature? Would be highly appreciated :)